lexos/processors/analyze/content_analysis_model.py

```diff
@@ -0,0 +1,488 @@
+                                             label=label))
+
+    def get_active_dicts(self) -> list:
+        """Gets a list containing all active dictionaries.
+
+        :return: a list containing all active dictionaries
+        """
+        return [dictionary for dictionary in self.dictionaries
+                if dictionary.active]
+
+    def count(self):
+        """Counts all dictionaries for all active files in the corpus."""
+        self._counters = []
+        dictionaries = self.join_active_dicts()
+        for file in deepcopy(self._corpus):
+            dictionaries = count_phrases(dictionaries, file)
+            self.get_dictionary_counts(dictionaries)
+
+    def get_dictionary_counts(self, dictionaries):
+        """Gets the counts for each dictionary.
+
+        :param dictionaries: list of Phrase object
+        """
+        counter = []
+        active_dicts = self.get_active_dicts()
+        for dictionary in active_dicts:
+            count = 0
+            for phrase in dictionaries:
+                if phrase.dict_label == dictionary.label:
+                    count += phrase.count
+            counter.append(count)
+            if len(counter) == len(active_dicts):
+                self._counters.append(counter)
+
+    def generate_scores(self):
+        """Calculate the formula and scores for each file in the corpus.
+
+        scores=formula/total_word_count
+        """
+        self._scores = []
+        self._formulas = []
+        active_dicts = self.get_active_dicts()
+        result = 0
+        for i, file in enumerate(self._corpus):
+            new_formula = self._formula
+            for j, active_dict in enumerate(active_dicts):
+                new_formula = new_formula.replace(
+                    "[" + active_dict.label + "]",
+                    str(self._counters[i][j]))
+            new_formula = new_formula.replace("()", "")
+            try:
+                result = eval(new_formula)
+            except (ValueError, SyntaxError):
+                pass
+            self._scores.append(round(
+                float(result) / file.total_word_count, ndigits=3))
+            self._formulas.append(result)
+
+    def generate_averages(self):
+        """Calculates the averages of every row in the table."""
+        self._averages = []
+        scores_sum = 0
+        total_word_counts_sum = 0
+        formulas_sum = 0
+        active_dicts = self.get_active_dicts()
+        for index, (score, formula, file) in enumerate(zip(self.scores,
+                                                           self._formulas,
+                                                           self._corpus)):
+            scores_sum += score
+            total_word_counts_sum += \
+                file.total_word_count
+            formulas_sum += formula
+        if len(self.scores) != 0:
+            scores_avg = round(
+                (float(scores_sum) / len(self.scores)), ndigits=3)
```

Comment:
in this function you are rounding couple of times. Is there a particular reason that you sometimes round to 3 digits, and sometimes round to 1? 
And since your are rounding to 1 digits couple of times, is it possible that you can create a name for it, to avoid it being a magic number?

Reply:
Yes, I round to 1 decimal place the averages of the raw counts for each dictionary, and to 3 decimal places the score (= formula result / total for count of the file) which is a smaller number. An alternative to rounding the score to 3 decimal places could be to convert scores to percentages.

Reply:
In that case I think you need to make couple of constant for them to avoid being magic numbers